packages/jest-message-util/src/visitor.ts

```diff
@@ -0,0 +1,158 @@
+    const header = this.ErrorHeader(error);
+    const frames = this.Frames(error.frames);
+
+    return frames && header ? `${header}\n\n${frames}` : header || frames;
+  }
+
+  ErrorHeader(error: ErrorNode): string | undefined {
+    const {options} = this;
+    const name = this.visit(error.name);
+    const name_ = name === 'Error' ? undefined : name;
+    const message = error.message ? this.visit(error.message) : '';
+    const message_ = !name && message === 'Error' ? undefined : message;
+    const header =
+      message_ && name_ ? `${name_}: ${message_}` : name_ || message_;
+    return header && indentLines(header, options.indentHeader);
+  }
+
+  Frames(frames: Array<FrameNode> | undefined): string | undefined {
+    const {options} = this;
+    if (!frames || !options.frames) return undefined;
+
+    let codeFrame;
+    if (!options.noCodeFrame) {
+      const topSourceFrame = getTopFrame(frames);
+
+      const absPath = topSourceFrame && getAbsoluteSitePath(topSourceFrame);
+      if (topSourceFrame && absPath) {
+        const {site} = topSourceFrame.callSite;
+
+        if (site.position) {
+          const {line, column} = site.position;
+          let fileContent;
+          try {
+            // TODO: check & read HasteFS instead of reading the filesystem:
+            // see: https://github.com/facebook/jest/pull/5405#discussion_r164281696
+            fileContent = fs.readFileSync(absPath, 'utf8');
+            codeFrame = codeFrameColumns(
+              fileContent,
+              {start: {column, line}},
+              {highlightCode: true},
+            );
+          } catch {
+            // the file does not exist or is inaccessible, we ignore
+          }
+        }
+      }
+    }
+
+    const printedFrames = frames.map(frame => this.visit(frame)).join('\n');
+
+    const printed =
+      printedFrames && codeFrame
+        ? `${codeFrame || ''}\n\n${printedFrames}`
+        : codeFrame || printedFrames || '';
+
+    return printed && indentLines(printed, options.indentFrames);
+  }
+
+  Frame(frame: FrameNode): string {
+    return STACK_TRACE_COLOR(super.Frame(frame).slice(2));
+  }
+
+  PathLocator(locator: PathLocatorNode): string {
+    const {options} = this;
+    const {path} = locator;
+    if (options.rootDir && path.startsWith('/')) {
+      const relPath = relative(options.rootDir, path);
+      return relPath;
+    }
+    return super.PathLocator(locator);
+  }
+
+  Site(site: SiteNode): string {
+    const {options} = this;
+    const {testMatch} = options;
+
+    const printed = super.Site(site);
+    const absPath = getAbsoluteSitePath(site);
+    if (absPath) {
+      const projPath = relative(options.rootDir, absPath);
+
+      const isTestPath =
+        (testMatch &&
+          testMatch.length &&
+          micromatch([projPath], testMatch).length > 0) ||
+        projPath === options.relativeTestPath;
+
+      if (isTestPath) return chalk.cyan(printed);
+    }
+    return printed;
+  }
+}
```

Comment:
I probably want to export and use a `printNode(node)` function, just like I do in `stack-tools` itself.